migrations/20190410092014_publication_tag.js

```diff
@@ -9,7 +9,8 @@ exports.up = function(knex, Promise) {
       .notNullable()
       .onDelete('CASCADE')
       .index()
-    table.string('tagId').references('id').inTable('Tag').notNullable().onDelete('CASCADE').index();
+    table.string('tagId').references('id').inTable('Tag').notNullable().onDelete('CASCADE').index()
+    table.unique(['publicationId', 'tagId'])
```

Comment:
you should do the same with note_tag. We don't use it yet but we should avoid having to change migration files later when we can.